openedx/core/djangoapps/content/search/documents.py

```diff
@@ -546,3 +547,66 @@ def searchable_doc_for_collection(
             doc['_disabled'] = True
 
     return doc
+
+
+def searchable_doc_for_container(
+    container_key: LibraryContainerLocator,
+) -> dict:
+    """
+    Generate a dictionary document suitable for ingestion into a search engine
+    like Meilisearch or Elasticsearch, so that the given collection can be
+    found using faceted search.
+
+    If no container is found for the given container key, the returned document
+    will contain only basic information derived from the container key, and no
+    Fields.type value will be included in the returned dict.
+    """
+    doc = {
+        Fields.id: meili_id_from_opaque_key(container_key),
+        Fields.context_key: str(container_key.library_key),
+        Fields.org: str(container_key.org),
+        # In the future, this may be either course_container or library_container
+        Fields.type: DocType.library_container,
+        # To check if it is "unit", "section", "subsection", etc..
+        Fields.block_type: container_key.container_type,
+        Fields.usage_key: str(container_key),  # Field name isn't exact but this is the closest match
+        Fields.block_id: container_key.container_id,  # Field name isn't exact but this is the closest match
+        Fields.access_id: _meili_access_id_from_context_key(container_key.library_key),
+    }
```

Comment:
I think `publish_status` should be set to `PublishStatus.Never`even if we don't have a container (same as how `searchable_doc_for_library_block` does):
```suggestion
        Field.publish_status: PublishStatus.never,
    }
```

Reply:
Updated: https://github.com/openedx/edx-platform/pull/36432/commits/c3406d895cb7cbd41ac88beaeff1aae3c9261ced Thanks!